pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -169,12 +170,7 @@ public Consumer(Subscription subscription, SubType subType, String topicName, lo
         stats.setClientVersion(cnx.getClientVersion());
         stats.metadata = this.metadata;
 
-        if (Subscription.isIndividualAckMode(subType)) {
-            this.pendingAcks = new ConcurrentLongLongPairHashMap(256, 1);
-        } else {
-            // We don't need to keep track of pending acks if the subscription is not shared
-            this.pendingAcks = null;
-        }
+        this.pendingAcks = new ConcurrentLongLongPairHashMap(256, 1);
     }
```

Comment:
Can we avoid creating a hashmap if we don't need it?

Reply:
we need to track of pending acks in all subscription mode, I thinks this is also need to create a hashmap

Reply:
pending acks are only used when the transaction is enabled.